paperslicer/media/extractor.py

```
@@ -0,0 +1,93 @@
+            page = doc[page_no - 1]
+            rect = fitz.Rect(*bbox)
+            pix = page.get_pixmap(clip=rect, dpi=300)
+            if it.get("type") == "table":
+                tab_i += 1
+                img_name = f"table_{tab_i:02d}.png"
+            else:
+                fig_i += 1
+                img_name = f"fig_{fig_i:02d}.png"
+            img_path = os.path.join(outdir, img_name)
+            pix.save(img_path)
+            sha1 = hashlib.sha1(open(img_path, "rb").read()).hexdigest()[:8]
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)